v1/server/server.go

```diff
@@ -1148,19 +1148,23 @@ func (s *Server) v0QueryPath(w http.ResponseWriter, r *http.Request, urlPath str
 	}
 
 	if len(rs) == 0 {
-		ref := stringPathToDataRef(urlPath)
+		ref, err := stringPathToDataRef(urlPath)
+		if err != nil {
+			writer.Error(w, http.StatusBadRequest, types.NewErrorV1(types.CodeUndefinedDocument, "invalid path: %v", err))
```

Comment:
Let's maybe also use types.CodeInvalidParameter here? It's a user-provided parameter, after all. Undefined document to me seems related to a policy being unexpectedly absent.